pkg/resources/schema.go

```diff
@@ -29,268 +33,380 @@ var schemaSchema = map[string]*schema.Schema{
+		return diag.FromErr(err)
 	}
 
-	if opts := s.Options; opts != nil && *opts != "" {
-		for _, opt := range strings.Split(*opts, ", ") {
-			switch opt {
-			case "TRANSIENT":
-				if err := d.Set("is_transient", true); err != nil {
-					return err
-				}
-			case "MANAGED ACCESS":
-				if err := d.Set("is_managed", true); err != nil {
-					return err
-				}
-			}
-		}
+	if diags := HandleDatabaseParameterRead(d, schemaParameters); diags != nil {
+		return diags
+	}
+	pipeExecutionPaused, err := collections.FindOne(schemaParameters, func(property *sdk.Parameter) bool {
+		return property.Key == "PIPE_EXECUTION_PAUSED"
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to find schema PIPE_EXECUTION_PAUSED parameter, err = %w", err))
+	}
+	value, err := strconv.ParseBool((*pipeExecutionPaused).Value)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("pipe_execution_paused", value); err != nil {
+		return diag.FromErr(err)
 	}
 
+	if err = d.Set(ShowOutputAttributeName, []map[string]any{schemas.SchemaToSchema(schema)}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(ParametersAttributeName, []map[string]any{schemas.SchemaParametersToSchema(schemaParameters)}); err != nil {
+		return diag.FromErr(err)
+	}
 	return nil
 }
 
-// UpdateSchema implements schema.UpdateFunc.
-func UpdateSchema(d *schema.ResourceData, meta interface{}) error {
+func UpdateContextSchema(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.DatabaseObjectIdentifier)
 	client := meta.(*provider.Context).Client
-	ctx := context.Background()
 
 	if d.HasChange("name") {
-		newId := sdk.NewDatabaseObjectIdentifier(id.DatabaseName(), d.Get("name").(string))
-
+		newId := sdk.NewDatabaseObjectIdentifier(d.Get("database").(string), d.Get("name").(string))
 		err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
 			NewName: sdk.Pointer(newId),
 		})
 		if err != nil {
-			return fmt.Errorf("error updating schema name on %v err = %w", d.Id(), err)
+			return diag.FromErr(err)
 		}
-
 		d.SetId(helpers.EncodeSnowflakeID(newId))
 		id = newId
 	}
 
-	if d.HasChange("comment") {
-		comment := d.Get("comment")
-		if comment != "" {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				Set: &sdk.SchemaSet{
-					Comment: sdk.String(comment.(string)),
-				},
-			})
-			if err != nil {
-				return fmt.Errorf("error updating schema comment on %v err = %w", d.Id(), err)
-			}
-		} else {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				Unset: &sdk.SchemaUnset{
-					Comment: sdk.Bool(true),
-				},
-			})
-			if err != nil {
-				return fmt.Errorf("error updating schema comment on %v err = %w", d.Id(), err)
-			}
-		}
-	}
-
-	if d.HasChange("is_managed") {
-		managed := d.Get("is_managed")
+	if d.HasChange("with_managed_access") {
 		var err error
-		if managed.(bool) {
+		if _, ok := d.GetOk("with_managed_access"); ok {
```

Comment:
Shouldn't this be used with the `d.Get` and compared with the BooleanTrue value?

Reply:
Why? This is a regular Bool, not BooleanString.

Reply:
Changed to BooleanString - done.